### PR TITLE
Feature metrics shipping

### PR DESCRIPTION
**Pull request by simonasr**
Add feature to configure puppet server metrics (manage metrics.conf).

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -685,6 +702,13 @@
   Integer[0] $server_web_idle_timeout = $puppet::params::server_web_idle_timeout,
   Boolean $server_puppetserver_jruby9k = $puppet::params::server_puppetserver_jruby9k,
   Boolean $server_puppetserver_metrics = $puppet::params::server_puppetserver_metrics,
+  Optional[Boolean] $server_metrics_jmx_enable = $::puppet::params::server_metrics_jmx_enable,
+  Optional[Boolean] $server_metrics_graphite_enable = $::puppet::params::server_metrics_graphite_enable,
+  Optional[String] $server_metrics_graphite_host = $::puppet::params::server_metrics_graphite_host,
+  Optional[Integer] $server_metrics_graphite_port = $::puppet::params::server_metrics_graphite_port,
+  Optional[String] $server_metrics_server_id = $::puppet::params::server_metrics_server_id,
+  Optional[Integer] $server_metrics_graphite_interval = $::puppet::params::server_metrics_graphite_interval,
+  Optional[Variant[Undef,Array]] $server_metrics_allowed = $::puppet::params::server_metrics_allowed,
```
Comment (ekohl) [inline manifests/init.pp]: Optional and Undef is redundant.

File: manifests/server/puppetserver.pp [hunk 2]
```diff
@@ -256,6 +263,13 @@
     content => template('puppet/server/puppetserver/conf.d/auth.conf.erb'),
   }
 
+  if versioncmp($server_puppetserver_version, '5.0') >= 0 and $server_metrics {
+    file { "${server_puppetserver_dir}/conf.d/metrics.conf":
```
Comment (ekohl) [inline manifests/server/puppetserver.pp]: Do you want to ensure it's absent otherwise?

**Comment by mmoll** [comment]
Yes, -hocon should be used now, GH-537 can serve as an example.

**Comment by simonasr** [comment]
Adjusted code to use puppetlabs-hocon for managing metrics.conf. Thank you for pointing that out :)

File: manifests/server/puppetserver.pp [hunk 3]
```diff
@@ -348,4 +355,50 @@
   file { $product_conf:
     ensure => $product_conf_ensure,
   }
+
+  if versioncmp($server_puppetserver_version, '5.0') >= 0 {
```
[the remark this answers is not in the document]
Comment (simonasr) [reply manifests/server/puppetserver.pp]: Older versions ignore this file. "Performance metrics" (and metrics.conf) was introduced in Puppet 5 platform

File: manifests/init.pp [hunk 4]
```diff
@@ -685,6 +702,13 @@
   Integer[0] $server_web_idle_timeout = $puppet::params::server_web_idle_timeout,
   Boolean $server_puppetserver_jruby9k = $puppet::params::server_puppetserver_jruby9k,
   Boolean $server_puppetserver_metrics = $puppet::params::server_puppetserver_metrics,
+  Optional[Boolean] $server_metrics_jmx_enable = $::puppet::params::server_metrics_jmx_enable,
```
[the remark this answers is not in the document]
Comment (simonasr) [reply manifests/init.pp]: thanks, removed

File: manifests/params.pp [hunk 5]
```diff
@@ -475,6 +475,15 @@
   # this switch also controls Ruby profiling, by default disabled for Puppetserver 2.x, enabled for 5.x
   $server_puppetserver_metrics = versioncmp($::puppetversion, '5.0.0') >= 0
 
+  # Puppetserver metrics shipping
+  $server_metrics_jmx_enable        = true
+  $server_metrics_graphite_enable   = true
```
Comment (mmoll) [inline manifests/params.pp]: I'm unsure about this. Doesn't that mean, per default puppetserver will try to send its data to that graphite host?

Comment (simonasr) [reply manifests/params.pp]: I think you are right. Changed $server_metrics_graphite_enable default value to false
 and left $server_metrics_jmx_enable unchanged. Now these settings are same as defaults in original metrics.conf.

**Comment by mmoll** [comment]
merged, ačiū @simonasr!

**Comment by simonasr** [comment]
Cool! @mmoll and @ekohl thank you very much for your support! :)

**Pull request merged**